arch/src/main/java/knaufdan/android/arch/base/component/IComponent.kt

```diff
@@ -20,4 +20,12 @@ interface IComponent<DataSource> : IGenericType<DataSource> {
      * @return [DataSource] which should be bound into [LayoutRes].
      */
     fun getDataSource(): DataSource
+
+    fun getId(): String = String.format(
+        "%s_ID :: %d_%s_%d",
+        this::class.simpleName,
+        hashCode(),
+        getDataSource(),
+        getDataSource().hashCode()
+    )
```

Comment:
- Remove `getDataSource`, just the two hashCodes should be unique. --> `%s_ID: %d_%d`

- Use Kotlin String construction to make it more readable: `${this::class.simpleName}_ID :: ${this.hashCode()}_${getDataSource().hashCode()}`

Reply:
done.